Source/MQTTnet/Adapter/IMqttChannelAdapter.cs

```diff
@@ -21,17 +21,15 @@ public interface IMqttChannelAdapter : IDisposable
 
         long BytesReceived { get; }
 
-        Action ReadingPacketStartedCallback { get; set; }
-
-        Action ReadingPacketCompletedCallback { get; set; }
+        bool IsReadingPacket { get; }
 
         Task ConnectAsync(TimeSpan timeout, CancellationToken cancellationToken);
 
         Task DisconnectAsync(TimeSpan timeout, CancellationToken cancellationToken);
```

Comment:
can we also remove the timeout for all the apis and just use the cancellation token?

Reply:
I was also thinking about this already. I would say yes but not yet since is this an official feature. But I have it in my notes.